File: ruby/raindrops/raindrops.rb
class Raindrops

  DROPS = 
   { 3 => "Pling",
     5 => "Plang",
     7 => "Plong"
   }

  def self.convert(num)
    result = DROPS.map do |num_key, word|
      word if num % num_key == 0
    end.join
    result.empty? ? num.to_s : result
  end

end

File: javascript/raindrops/raindrops.js
function Raindrops () {
}

Raindrops.prototype.convert = function (num) {
  var result = ""; 

  if (num % 3 === 0) {
    result = result + 'Pling';
  }

  if (num % 5 === 0) {
    result = result + 'Plang';
  }
  
  if (num % 7 === 0) {
    result = result + 'Plong';
  }

  return result ? result : num.toString();
};

module.exports = Raindrops;